src/main/java/com/gtnh/findit/service/itemfinder/FindItemRequest.java

```diff
@@ -1,106 +1,102 @@
 package com.gtnh.findit.service.itemfinder;
 
-import java.util.Optional;
-
+import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraftforge.fluids.FluidRegistry;
 import net.minecraftforge.fluids.FluidStack;
 
 import com.gtnh.findit.FindIt;
 import com.gtnh.findit.FindItConfig;
+import com.gtnh.findit.IStackFilter;
+import com.gtnh.findit.IStackFilter.IStackFilterProvider;
 import com.gtnh.findit.service.blockfinder.BlockFoundResponse;
 import com.gtnh.findit.util.ProtoUtils;
-import com.gtnh.findit.util.mods.ForestryUtils;
 
 import codechicken.nei.recipe.StackInfo;
 import cpw.mods.fml.common.network.simpleimpl.IMessage;
 import cpw.mods.fml.common.network.simpleimpl.IMessageHandler;
 import cpw.mods.fml.common.network.simpleimpl.MessageContext;
+import crazypants.enderio.conduit.TileConduitBundle;
+import gregtech.api.metatileentity.BaseMetaPipeEntity;
 import io.netty.buffer.ByteBuf;
 
 public class FindItemRequest implements IMessage {
 
     private ItemStack targetStack;
     private FluidStack targetFluidStack;
+    private boolean highlightingEmptyItemStacks;
+    private boolean searchInGTPipes;
+    private boolean searchConduits;
 
     public FindItemRequest(ItemStack targetStack) {
         this.targetStack = targetStack;
         this.targetFluidStack = StackInfo.getFluid(targetStack);
+        this.highlightingEmptyItemStacks = FindItConfig.ITEM_HIGHLIGHTING_EMPTY_ITEMSTACKS;
+        this.searchInGTPipes = FindItConfig.SEARCH_IN_GT_PIPES;
+        this.searchConduits = FindItConfig.SEARCH_IN_ENDERIO_CONDUITS;
```

Comment:
Storing and syncing of all these values are performed for respecting client config, am I correctly understanding your logic?

Reply:
yes